src/lua/LuaSpace.cpp

```diff
@@ -267,7 +267,7 @@ static int l_space_put_ship_on_route(lua_State *l)
 	const int coll = CheckCollision(ship, reldir, targdist, targpos, 0, erad);
 	if (coll) {
 		// need to correct positon, to avoid collision
-		if (Frame::GetFrame(shipFrame->GetNonRotFrame()) != Frame::GetFrame(targFrame->GetNonRotFrame()) && targpos.Length() > erad) {
+		if (shipFrame->GetNonRotFrame() != targFrame->GetNonRotFrame() && targpos.Length() > erad) {
 			// the ship is not in the target's frame or target is above the effective radius of obstructor - rotate the ship's position
 			// around the target position, so that the obstructor's "effective radius" does not cross the path
 			// direction obstructor -> target
```

Comment:
Interesting dillemma here, the code checks that we are both not in the same frame and further than the effective radius, however the comment is saying that this is an 'or' condition - what happens if we are in the same frame but beyond effective radius?

Reply:
@Web-eWorks Really weird. After paradoxically long deliberation, it seems to me that we need to change the comment to "and", because we want to enter the sphere of effective radius of obstructor only perpendicular to it, even if for some reason the target is not in the obstructor frame. (perhaps it could be an orbital station, in a very very low orbit?)

Reply:
@Web-eWorks if we are in the same frame but the target is beyond effective radius, like in any other case, when the target is beyond the effective radius, we can calculate the tangent, so option 1 can be used. I think I should remove the frame check at all.

Reply:
Feel free - I'm not particularly familiar with this algorithm, but that inconsistency stood out.

Reply:
@Web-eWorks done!